Remove feature flag for EmailVerificationGate

coreh opened the pull request:
This PR removes the feature flag for EmailVerificationGate, enabling it (and reactive email confirmation, and the post editor notice) everywhere.

Test live: https://calypso.live/?branch=enable/verification-gate-everywhere

coreh commented:
This also fixes most of the code quality issues pointed out by @aduth on #5475. For the passing of `user` and `userUtils` props, I'm opening a separate issue.

southp commented on client/post-editor/editor-ground-control/index.jsx:
```diff
@@ -67,28 +66,28 @@ export default React.createClass( {
 
 	componentDidMount: function() {
-		if ( ! config( 'email_verification_gate' ) ) return;
+		if ( !this.props.user ) {
```
Though I am aware of it will be gone in no time, I think it would look better if we add a space following `!` before we tackled #6256.

southp commented on client/post-editor/editor-ground-control/index.jsx:
```diff
@@ -67,28 +66,28 @@ export default React.createClass( {
 
 	componentWillUnmount: function() {
-		if ( ! config( 'email_verification_gate' ) ) return;
+		if ( !this.props.user ) {
```
Same here as above ^

southp commented on client/post-editor/verify-email-dialog.jsx:
```diff
@@ -43,6 +43,16 @@ class VerifyEmailDialog extends React.Component {
 		}.bind( this ) );
 	}
 
+	getResendbuttonLabel() {
```
From my understanding, "resend" and "button" are two separate words. If that's the case, we should put it as `getResnedButtonLabel` for consistency.

jordwest replied:
```diff
- getResnedButtonLabel
+ getResendButtonLabel
```
:)

southp commented:
Tested the following cases:

1. Logging in as a verified user, the warning was not appeared as expected.
1. Logging in as a unverified user, the warning was appeared as expected.
1. Resending the verification mail via the link contained, verified, and saw the warning box disappeared without refreshing.
1. Refreshing the page, and the waring was not appeared as expected.

 I've added some nit-picking comments, but it totally works great on my side. I am not the right person to give a more thorough structural opinion or give a shipping call, so I am calling out here @jordwest @dllh :)

jordwest commented on client/post-editor/verify-email-dialog.jsx:
```diff
@@ -43,6 +43,16 @@ class VerifyEmailDialog extends React.Component {
+		}
+		return i18n.translate( 'Resend Email' );
+	}
+
```
👍  much easier to follow than the bit it replaces

jordwest commented:
Looks like this needs a rebase. The code looks good to me, but I noticed there are some environments where the feature flag isn't already enabled - specifically the Desktop apps and production. Should the flag be turned on in each of those before removing all the feature flag conditionals?

coreh commented:
> I've added some nit-picking comments [...]

Fixed those, nice catch on the incorrect camel case name

> Should the flag be turned on in each of those before removing all the feature flag conditionals?

I don't think it's needed, since our goal is to enable it everywhere. (We tested it on wpcalypso.) First enabling it, then removing the check would be redundant, I think. Thoughts @michaeldcain?

> Looks like this needs a rebase. 

Ya, going to rebase it. Wonder what's conflicting?

michaeldcain commented:
LGTM!
